### ISSUE TITLE
No Op registered for Switch with domain_version of 11

### ISSUE THREAD

**Issue by florisdesmedt** [onnx/keras-onnx#471]
Hi,

I'm working on creating an onnx-model for efficientdet (https://github.com/xuannianz/EfficientDet). To make the `keras2onnx.convert_keras` call succeed I had to change lambda functions for the `swish` activations to regular tensorflow swish activations.
At last I obtained an onnx modelfile, but when loading this model file I get the error `[ONNXRuntimeError] : 10 : INVALID_GRAPH : This is an invalid model. Error in Node:block2b_drop/cond/Shape/Switch : No Op registered for Switch with domain_version of 11`

Anybody knows how to solve this?

Best regards,
F.

**Comment by wenbingl** [comment]
I think this node belongs to a dropout layer, I think it could be eliminated in ONNX graph. 
Can you share some code about which model in efficientDet source is to be converted? We could improve the converter a little.

**Comment by florisdesmedt** [comment]
Hi,
Thanks for the tip, I will look into eliminating the dropout layers from the model.
Currently I started with converting the first model (efficientdet with phi=0). I think once this one works it would not be that hard to apply the convertion to the other models also, since these duplicate parts of the network to improve accuracy.

`from model import efficientdet
m_train, model = efficientdet(phi=0,
                            weighted_bifpn=True,
                            num_classes=90,
                            score_threshold=0.3)
# load weights from (downloaded) coco weights
model_path = "efficientdet-d0.h5"
m_train.load_weights(model_path, by_name=True)

# convert to ONNX
import os
os.environ["TF_KERAS"] = "1"
import keras2onnx
output_model_path = "keras_efficientDet0.onnx"
onnx_model = keras2onnx.convert_keras(m_train, m_train.name)
keras2onnx.save_model(onnx_model, output_model_path)
`
As you see at the moment I focus on converting the train model. The prediction model contains additional operations to convert the results to actual detections (e.g. non-maximum suppression).

**Comment by wenbingl** [comment]
@jiafatom , Can you take a look at this issue?

**Comment by jiafatom** [comment]
The converter cannot extract the function from the lambda in Activation layer `layers.Activation(lambda x: tf.nn.swish(x))` in efficientdet. 
Please replace it as `layers.Lambda(lambda x: tf.nn.swish(x))`

**Comment by jiafatom** [comment]
@florisdesmedt We have some fixes recently. Can you pull latest keras2onnx master and retry? As I mentioned, you need make changes in original keras model, that is replace every `layers.Activation(lambda x: tf.nn.swish(x))` with `layers.Lambda(lambda x: tf.nn.swish(x))`. After this change, I tested it working well on onnxruntime.
